src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs

```diff
@@ -5968,6 +5968,309 @@ static void validate(ModuleSymbol module)
             }
         }
 
+        [ConditionalFact(typeof(CoreClrOnly))]
+        [WorkItem(57325, "https://github.com/dotnet/roslyn/issues/57325")]
+        public void InOutAttributes()
+        {
+            var source = @"
+using System.Runtime.InteropServices;
+
+class State
+{
+    public bool B;
+}
+
+class Program
+{
+    static void M([In] [Out] State state)
+    {
+        local(state);
+
+        static void local([In] [Out] State state)
+        {
+        }
+    }
+}
+";
+            var verifier = CompileAndVerify(
+                source,
+                targetFramework: TargetFramework.StandardAndCSharp,
+                symbolValidator: validateMetadata,
+                options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All));
+            var methodParam = ((CSharpCompilation)verifier.Compilation).GetMember<MethodSymbol>("Program.M").Parameters[0];
+            Assert.True(methodParam.IsMetadataIn);
+            Assert.True(methodParam.IsMetadataOut);
+
+            var localFunctionParam = verifier.FindLocalFunction("local").Parameters[0].GetSymbol<ParameterSymbol>();
+            Assert.True(localFunctionParam.IsMetadataIn);
+            Assert.True(localFunctionParam.IsMetadataOut);
+
+            verifier.VerifyTypeIL("Program", """
```

Comment:
>VerifyTypeIL [](http://example.com/codeflow?start=21&length=12)

What is the goal to use ```VerifyTypeIL```? Is asserting ```IsMetadataIn/IsMetadataOut``` insufficient? Otherwise, consider removing IL comparison.